Fix #14240: Blank link shown in theme actions dropdown

pedroasgomes opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✔️
| New feature/enhancement? (use the a.x branch)      | ❌
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ✔️
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #14240 



## Description

This PR fixes an issue where the dropdown menu for individual non-system themes showed a blank link in the list of available actions. The issue was caused by empty button definitions being passed to the `customButtons` array in the Twig template.

To fix this, the `list.html.twig` file was updated to ensure that only properly defined buttons are merged into the `customButtons` array. Buttons such as the preview or visibility toggles are now conditionally added only when they are not empty, preventing malformed entries from being rendered.

An E2E test is included to ensure that no action link with `data-toggle="ajax"` is rendered without a corresponding `href` attribute.

| Before                                 | After
| -------------------------------------- | ---
| ![Before](https://github.com/user-attachments/assets/4536894f-ebf7-4db5-aee5-8201fac2709e) | ![After](https://github.com/user-attachments/assets/fd08095c-dfef-4e3a-81e4-aa3a6d4e6cf2)  |


---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Navigate to the **Themes** Page
3. Locate a theme that is **not set as system theme** (that can be removed).
4. Click the dropdown for actions.
5. Confirm the result

escopecz commented:
Hey! Thanks for this fix! It's amazing!

1. I think we should find why is there the empty option rather than ignoring empty options. I think the empty option happens here: https://github.com/mautic/mautic/blob/7.x/app/bundles/CoreBundle/Resources/views/Theme/list.html.twig#L33. It shouldn't be added to the customButtons array if it's empty.
2. The test checks if there is a link with `data-toggle="ajax"` attribute. How about to check rather if the link has a `href` attribute and if it doesn't then fail the test? That could be more helpful.

pedroasgomes committed:
Fix #14240: Actions show blank link for themes

When clicking to open the list dropdown for a theme on the Themes page,
blank links are shown at the top and it happens only for themes that are
not set as system theme.
To fix this, Resources/views/Theme/list.html.twig was updated to prevent
empty button definitions from being added to the `customButtons` array.
Now, only buttons with proper attributes are merged into the list of
actions shown in the dropdown menu.
An E2E test is included to verify that no links with
`data-toggle="ajax"` are rendered without a `href` attribute.

pedroasgomes commented:
Hi @escopecz, just following up — the changes were pushed over the weekend:

- Empty buttons are now filtered before being added to `customButtons` in the Twig template
- An E2E test ensures that no `data-toggle="ajax"` link is rendered without a `href`

Let me know if this aligns with what you had in mind, and I’m happy to adjust if needed. Appreciate your time!

escopecz reviewed (dismissed):
The code changes look great! And I tested it and it fixes the empty menu item.
![Screenshot 2025-04-07 at 10 41 18](https://github.com/user-attachments/assets/a25ecd40-5431-43ee-b588-f365b0a0ec01)


Thank you!

escopecz commented:
One last thing is that the CS Fixer is failing. Please run
```
bin/php-cs-fixer fix --config=.php-cs-fixer.php --using-cache=no --show-progress=dots --diff tests/_support/Page/Acceptance/ThemesPage.php tests/acceptance/ThemeManagementCest.php
```
To fix these issues.

pedroasgomes committed:
Fix #14240: Apply CS fixes to the E2E test files

Fixes PHP CS violations flagged during review. No
logic or structure changes were made — only style
adjustments in accordance with the
`.php-cs-fixer.php` config.

escopecz reviewed (approved):
Thank you! The changes looks good, and works as described 👍

escopecz commented:
@all-contributors please add @pedroasgomes for code
